### PR TITLE
[new] Add mapbox-gl via single package.json w/ npm auto-update

### PR DESCRIPTION
**Pull request by jasonbarry**
Pull request for issue: #10654 

# Profile of the lib
 * Git repository (required): https://github.com/mapbox/mapbox-gl-js
 * Official website (optional, not the repository): https://www.mapbox.com/mapbox-gl-js/api/
 * NPM package url (optional): https://www.npmjs.com/package/mapbox-gl
 * License and its reference: [BSD-3-Clause](https://www.npmjs.com/package/mapbox-gl)
 * GitHub / Bitbucket popularity (required): 
   - Count of watchers: 221
   - Count of stars: 1554
   - Count of forks: 367
 * NPM download stats (optional):
   - Downloads in the last day: 962
   - Downloads in the last week: 11352
   - Downloads in the last month: 46516

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/mapbox-gl/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "url": "git://github.com/mapbox/mapbox-gl-js.git"
+  }, 
+  "npmName": "mapbox-gl",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "dist/mapbox-gl.css",
+        "src/index.js"
+      ]
```
Comment (PeterDaveHello) [inline ajax/libs/mapbox-gl/package.json]: We should use `dist` and `src` as `basePath`.

File: ajax/libs/mapbox-gl/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "mapbox-gl",
+  "description": "A WebGL interactive maps library",
+  "version": "0.32.1",
+  "main": "src/index.js",
```
Comment (PeterDaveHello) [inline ajax/libs/mapbox-gl/package.json]: Please remove `main` here.

File: ajax/libs/mapbox-gl/package.json (hunk 1 above)
Comment (jasonbarry) [inline ajax/libs/mapbox-gl/package.json]: What is the format for adding multiple base paths? Would it be `"basePath": ["dist", "src"]`?

Comment (PeterDaveHello) [inline ajax/libs/mapbox-gl/package.json]: You can replica the whole filemap object inside npmFileMap array:
```
{
  "basePath": "dist",
  files": [
    "**/*"
  ]
}
```

Comment (pvnr0082t) [inline ajax/libs/mapbox-gl/package.json]: I found that there is no `src` directory in NPM package, so I think we can just set `dist` as basePath.

File: ajax/libs/mapbox-gl/package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "mapbox-gl",
+  "description": "A WebGL interactive maps library",
+  "version": "0.32.1",
```
Comment (pvnr0082t) [inline ajax/libs/mapbox-gl/package.json]: Let's remove version field when adding a new lib via single package.json. Thanks.

File: ajax/libs/mapbox-gl/package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  "main": "src/index.js",
+  "filename": "src/index.min.js",
```
Comment (pvnr0082t) [inline ajax/libs/mapbox-gl/package.json]: `filename` should point to the minified main file. In this case, it should point to `mapbox-gl.js`.

File: ajax/libs/mapbox-gl/package.json (hunk 1 above)
Comment (maruilian11) [inline ajax/libs/mapbox-gl/package.json]: @pvnr0082t I think we also need files under `plugins`, how do you think?

Comment (jasonbarry) [inline ajax/libs/mapbox-gl/package.json]: @maruilian11 I think the plugins ([listed here](https://www.mapbox.com/mapbox-gl-js/plugins/)) should be a separate pull request, as they are separate packages available on npm (e.g. [mapbox-gl-draw](https://www.npmjs.com/package/@mapbox/mapbox-gl-draw)).

**Comment by jasonbarry** [comment]
Will fix these issues in a follow-up PR.

File: ajax/libs/mapbox-gl/package.json (hunk 1 above)
Comment (maruilian11) [inline ajax/libs/mapbox-gl/package.json]: Thank you for your explanation.
